libwhisper/WhisperPeer.cpp

```diff
@@ -104,3 +114,13 @@ void WhisperPeer::noteNewMessage(h256 _h, Envelope const& _m)
 	Guard l(x_unseen);
 	m_unseen.insert(make_pair(rating(_m), _h));
 }
+
+void WhisperPeer::sendTopicsOfInterest(FixedHash<TopicBloomFilterSize> const& _bloom)
+{
+	RLPStream s;
+	prep(s, TopicFilterPacket, 1);
+	s << _bloom;
+	sealAndSend(s);
+
+	m_advertiseTopicsOfInterest = false;
```

Comment:
m_advertiseTopicsOfInterest may be set to true while this method is executing and it will overwrite the value to false. Would be good to mutex m_advertiseTopicsOfInterest and set to false before sending the packet.